dot/network/block_announce.go

```diff
@@ -18,7 +18,7 @@ import (
 var errInvalidRole = errors.New("invalid role")
 var (
 	_ NotificationsMessage = &BlockAnnounceMessage{}
-	_ NotificationsMessage = &BlockAnnounceHandshake{}
+	_ Handshake            = &BlockAnnounceHandshake{}
```

Comment:
out of scope nit, but you can change to have zero allocation with

```suggestion
	_ Handshake            = (*BlockAnnounceHandshake)(nil)
```